lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -107,25 +107,33 @@ def upload_binary(app_id, binary_path)
       #
       # Returns the release_id on a successful release, otherwise returns nil.
       #
-      # Throws an error if the number of polling retries exceeds MAX_POLLING_RETRIES
+      # Throws a UI error if the number of polling retries exceeds MAX_POLLING_RETRIES
+      # or if it was not able to upload the binary
       def upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
         upload_status_response = get_upload_status(app_id, upload_token)
-        if upload_status_response.success?
+        if upload_status_response.success? || upload_status_response.already_uploaded?
           UI.success("This APK/IPA has been uploaded before. Skipping upload step.")
         else
-          UI.message("This APK has not been uploaded before.")
+          UI.message("This APK/IPA has not been uploaded before")
+          UI.message("Uploading the APK/IPA.")
```

Comment:
Seems like these messages should be inside the `unless`

Reply:
I think it's _okay_ if these are outside of the `unless` statement. For example: A developer uploads their binary, then for some reason they kill the current process. If they rerun it and the binary is still being processed in which case `upload_status_response.in_progress?` is `true`, if the message statements were inside of the `unless` statement it wouldn't display any messages.

Even though we _technically_ aren't uploading the binary again, I think it's better to show the messages tor provide information the developer. @lfkellogg thoughts?

Reply:
Yeah that makes sense, thanks. I think it's fine!